### ISSUE TITLE
Common Voice Dataset Training for other languages not giving good results - Word Error Rate (WER) Not Reducing - Stuck at High Values - 100%

### ISSUE THREAD

**Issue by aslon1213** [speechbrain/speechbrain#2810]
### Describe the bug

Hi everyone,

I am training a speech recognition model using SpeechBrain, but I noticed that while the loss is decreasing, my Word Error Rate (WER) remains high. Even after multiple epochs, it does not reduce below a certain threshold.

Training setup almost the same as example provided in recipes/CommonVoice/transformers/conformer-large.yaml except some data path and language specific parameters.

Training setup almost the same as example provided in recipes/CommonVoice/transformers/conformer-large.yaml:
	•	Model: transformers
	•	Dataset: CommonVoice corpus
Observations:
	•	Training and validation loss improve, but WER remains stuck at 100%.
	•	Increasing epochs does not help significantly.
	•	Model predictions contain a lot of substitutions/deletions/insertions.
	•	Some outputs are completely blank or garbage text.

Things I have tried:

✅ Adjusting learning rate
✅ Data augmentation (speed perturbation, noise addition)
✅ Checking label alignment in transcripts
✅ Changing beam search parameters in decoding
✅ Trying different architectures

Has anyone else faced this issue? Are there specific hyperparameters or decoding techniques that helped reduce WER in SpeechBrain?
I am using it for uzbek language of CommonVoice, which has about 400 hours of data.
Would appreciate any insights! Thanks.

### Expected behaviour

WER should lower as training

### To Reproduce

torchrun --standalone --nproc_per_node=6 python train.py hprams/conformer-large.yaml

### Environment Details

6 GPUS with 12GB RAM

### Relevant Log Output

```shell
Logs:
epoch: 1, lr: 8.32e-07, steps: 27 - train loss: 3.42e+02 - valid loss: 3.67e+02, valid ACC: 9.25e-05, valid WER: 1.62e+03, valid CER: 1.03e+03
epoch: 2, lr: 1.70e-06, steps: 54 - train loss: 2.82e+02 - valid loss: 2.51e+02, valid ACC: 1.05e-04, valid WER: 1.51e+03, valid CER: 1.05e+03
epoch: 3, lr: 2.56e-06, steps: 81 - train loss: 1.85e+02 - valid loss: 1.36e+02, valid ACC: 7.07e-02, valid WER: 2.65e+02, valid CER: 2.56e+02
epoch: 4, lr: 3.42e-06, steps: 108 - train loss: 1.17e+02 - valid loss: 1.13e+02, valid ACC: 7.07e-02, valid WER: 100.00, valid CER: 99.68
epoch: 5, lr: 4.29e-06, steps: 135 - train loss: 1.09e+02 - valid loss: 1.10e+02, valid ACC: 7.67e-02, valid WER: 1.00e+02, valid CER: 99.28
epoch: 6, lr: 5.15e-06, steps: 162 - train loss: 1.10e+02 - valid loss: 1.08e+02, valid ACC: 8.07e-02, valid WER: 1.00e+02, valid CER: 99.28
epoch: 7, lr: 6.02e-06, steps: 189 - train loss: 1.06e+02 - valid loss: 1.06e+02, valid ACC: 8.32e-02, valid WER: 1.00e+02, valid CER: 99.28
epoch: 8, lr: 6.88e-06, steps: 216 - train loss: 98.01 - valid loss: 1.05e+02, valid ACC: 8.38e-02, valid WER: 1.00e+02, valid CER: 99.28
epoch: 9, lr: 7.74e-06, steps: 243 - train loss: 1.09e+02 - valid loss: 1.04e+02, valid ACC: 8.42e-02, valid WER: 1.00e+02, valid CER: 99.28
epoch: 10, lr: 8.61e-06, steps: 270 - train loss: 1.02e+02 - valid loss: 1.03e+02, valid ACC: 8.62e-02, valid WER: 99.99, valid CER: 98.42
epoch: 11, lr: 9.47e-06, steps: 297 - train loss: 1.05e+02 - valid loss: 1.02e+02, valid ACC: 8.71e-02, valid WER: 100.00, valid CER: 97.57
epoch: 12, lr: 1.03e-05, steps: 324 - train loss: 96.42 - valid loss: 1.02e+02, valid ACC: 8.47e-02, valid WER: 100.00, valid CER: 97.19
epoch: 13, lr: 1.12e-05, steps: 351 - train loss: 96.79 - valid loss: 1.01e+02, valid ACC: 8.70e-02, valid WER: 100.00, valid CER: 97.20
epoch: 14, lr: 1.21e-05, steps: 378 - train loss: 96.04 - valid loss: 99.98, valid ACC: 9.07e-02, valid WER: 1.00e+02, valid CER: 97.51
```

**Comment by TParcollet** [comment]
Hi, you only do 27 steps per epoch, which means that the lr warmup will take forever. 1. Reduce the batch size by a lot; 2. Reduce a bit the warmup; 3. Remove data augmentation for now (to improve stability).

**Comment by aslon1213** [comment]
I did as you said and training got well at least by CER and WER values but when I do inference usign speechbrain.inference.ASR.EncoderDecoderASR module it is giving non sense output. 





# Contents of train_log.txt
epoch: 1, lr: 1.24e-04, steps: 928 - train loss: 98.15 - valid loss: 80.37, valid ACC: 2.29e-01
epoch: 2, lr: 2.47e-04, steps: 1856 - train loss: 66.02 - valid loss: 39.55, valid ACC: 5.96e-01
epoch: 3, lr: 3.71e-04, steps: 2784 - train loss: 27.12 - valid loss: 16.30, valid ACC: 8.59e-01
epoch: 4, lr: 4.95e-04, steps: 3712 - train loss: 14.93 - valid loss: 11.01, valid ACC: 9.01e-01
epoch: 5, lr: 6.19e-04, steps: 4640 - train loss: 11.02 - valid loss: 9.16, valid ACC: 9.16e-01
epoch: 6, lr: 7.42e-04, steps: 5568 - train loss: 9.18 - valid loss: 8.13, valid ACC: 9.23e-01
epoch: 7, lr: 7.69e-04, steps: 6496 - train loss: 8.00 - valid loss: 7.19, valid ACC: 9.31e-01
epoch: 8, lr: 7.19e-04, steps: 7424 - train loss: 6.79 - valid loss: 6.44, valid ACC: 9.38e-01
epoch: 9, lr: 6.78e-04, steps: 8352 - train loss: 13.98 - valid loss: 6.19, valid ACC: 9.42e-01
epoch: 10, lr: 6.43e-04, steps: 9280 - train loss: 21.76 - valid loss: 5.93, valid ACC: 9.45e-01, valid WER: 12.59, valid CER: 3.33
epoch: 11, lr: 6.20e-04, steps: 10000 - train loss: 19.60 - valid loss: 5.69, valid ACC: 9.48e-01


# These are params: 
sample_rate: 16000
n_fft: 400
n_mels: 80

# Model parameters
# Transformer
d_model: 384
nhead: 8
num_encoder_layers: 8
num_decoder_layers: 4
d_ffn: 1536
transformer_dropout: 0.1
activation: !name:torch.nn.GELU
output_neurons: 800

# Outputs
blank_index: 0
label_smoothing: 0.1
pad_index: 0
bos_index: 1
eos_index: 2

# Decoding parameters
min_decode_ratio: 0.0
max_decode_ratio: 1.0
valid_search_interval: 10
valid_beam_size: 1
test_beam_size: 10
ctc_weight_decode: 0.3
scorer_beam_scale: 0.3
transformer_beam_search: True

# Example Output: 
['TAQDIM MOSKVA TAQDIM MOSKVA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA MOSKVA TAQDIM ORASIDA TAQDIM MOSKVA TAQDIM MOSKVA TAQDIM MUNOSABAT TAQDIM MUNOSABAT TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA ORASIDA TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT RINBOSAR MUNOSABAT MOSKVA MUNOSABAT MOSKVA RINBOSAR MUNOSABAT MOSKVA MUNOSABAT MOSKVA TAQDIM ORASIDA TAQDIM MUNOSABAT ORASIDA MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT ORASIDA MUNOSABAT MOSKVA MUNOSABAT MOSKVA QISHLOQ MOSKVA TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT TAQDIM ORASIDA MUNOSABAT TAQDIM MUNOSABAT TAQDIM MUNOSABAT TAQDIM MUNOSABAT MOSKVA MUNOSABAT TAQDIM MUNOSABAT MOSKVA QIL TAQDIM ORASIDA HUJUM MUNOSABAT ORASIDA MUNOSABAT ORASIDA HUJUM MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUXLISLAR MUNOSABAT MOSKVA QISHLOQ TAQDIM MUNOSABAT ORASIDA TAQDIM ORASIDA MOSKVA HUJUM MOSKVA MUNOSABAT MOSKVA HUJUM TAQDIM MOSKVA MUNOSABAT ORASIDA RINBOSAR ORASIDA MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT TAQDIM ORASIDA TAQDIM MUNOSABAT MOSKVA QISHLOQ MOSKVA MUNOSABAT QISQA HUJUM MUNOSABAT ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA MOSKVA MUNOSABAT MOSKVA ORASIDA MOSKVA ORASIDA MOSKVA MUXLISLAR MOSKVA MUNOSABAT ORASIDA MUNOSABAT MOSKVA MUNOSABAT MOSKVA HUJUM MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT QIL MUNOSABAT BOSQICH QISQA RINBOSAR QISQA ORASIDA TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA ORASIDA TAQDIM MUNOSABAT ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM']
# SHOULD BE
OPASINI ONASINI SOCH QISTIRGICHLARINI LABIGA QISTIRIB O YNAGANLAR XAYRLI TONG

**Comment by aslon1213** [comment]
> Hi, you only do 27 steps per epoch, which means that the lr warmup will take forever. 1. Reduce the batch size by a lot; 2. Reduce a bit the warmup; 3. Remove data augmentation for now (to improve stability).

I did as you said and training got well at least by CER and WER values but when I do inference using speechbrain.inference.ASR.EncoderDecoderASR module it is giving non sense output.

# Contents of train_log.txt
epoch: 1, lr: 1.24e-04, steps: 928 - train loss: 98.15 - valid loss: 80.37, valid ACC: 2.29e-01
epoch: 2, lr: 2.47e-04, steps: 1856 - train loss: 66.02 - valid loss: 39.55, valid ACC: 5.96e-01
epoch: 3, lr: 3.71e-04, steps: 2784 - train loss: 27.12 - valid loss: 16.30, valid ACC: 8.59e-01
epoch: 4, lr: 4.95e-04, steps: 3712 - train loss: 14.93 - valid loss: 11.01, valid ACC: 9.01e-01
epoch: 5, lr: 6.19e-04, steps: 4640 - train loss: 11.02 - valid loss: 9.16, valid ACC: 9.16e-01
epoch: 6, lr: 7.42e-04, steps: 5568 - train loss: 9.18 - valid loss: 8.13, valid ACC: 9.23e-01
epoch: 7, lr: 7.69e-04, steps: 6496 - train loss: 8.00 - valid loss: 7.19, valid ACC: 9.31e-01
epoch: 8, lr: 7.19e-04, steps: 7424 - train loss: 6.79 - valid loss: 6.44, valid ACC: 9.38e-01
epoch: 9, lr: 6.78e-04, steps: 8352 - train loss: 13.98 - valid loss: 6.19, valid ACC: 9.42e-01
epoch: 10, lr: 6.43e-04, steps: 9280 - train loss: 21.76 - valid loss: 5.93, valid ACC: 9.45e-01, valid WER: 12.59, valid CER: 3.33
epoch: 11, lr: 6.20e-04, steps: 10000 - train loss: 19.60 - valid loss: 5.69, valid ACC: 9.48e-01

# These are params:
sample_rate: 16000
n_fft: 400
n_mels: 80

Model parameters
Transformer
d_model: 384
nhead: 8
num_encoder_layers: 8
num_decoder_layers: 4
d_ffn: 1536
transformer_dropout: 0.1
activation: !name:torch.nn.GELU
output_neurons: 800

Outputs
blank_index: 0
label_smoothing: 0.1
pad_index: 0
bos_index: 1
eos_index: 2

Decoding parameters
min_decode_ratio: 0.0
max_decode_ratio: 1.0
valid_search_interval: 10
valid_beam_size: 1
test_beam_size: 10
ctc_weight_decode: 0.3
scorer_beam_scale: 0.3
transformer_beam_search: True

# Example Output:
['TAQDIM MOSKVA TAQDIM MOSKVA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA MOSKVA TAQDIM ORASIDA TAQDIM MOSKVA TAQDIM MOSKVA TAQDIM MUNOSABAT TAQDIM MUNOSABAT TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA ORASIDA TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT RINBOSAR MUNOSABAT MOSKVA MUNOSABAT MOSKVA RINBOSAR MUNOSABAT MOSKVA MUNOSABAT MOSKVA TAQDIM ORASIDA TAQDIM MUNOSABAT ORASIDA MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT ORASIDA MUNOSABAT MOSKVA MUNOSABAT MOSKVA QISHLOQ MOSKVA TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT TAQDIM ORASIDA MUNOSABAT TAQDIM MUNOSABAT TAQDIM MUNOSABAT TAQDIM MUNOSABAT MOSKVA MUNOSABAT TAQDIM MUNOSABAT MOSKVA QIL TAQDIM ORASIDA HUJUM MUNOSABAT ORASIDA MUNOSABAT ORASIDA HUJUM MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUXLISLAR MUNOSABAT MOSKVA QISHLOQ TAQDIM MUNOSABAT ORASIDA TAQDIM ORASIDA MOSKVA HUJUM MOSKVA MUNOSABAT MOSKVA HUJUM TAQDIM MOSKVA MUNOSABAT ORASIDA RINBOSAR ORASIDA MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT TAQDIM ORASIDA TAQDIM MUNOSABAT MOSKVA QISHLOQ MOSKVA MUNOSABAT QISQA HUJUM MUNOSABAT ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA MOSKVA MUNOSABAT MOSKVA ORASIDA MOSKVA ORASIDA MOSKVA MUXLISLAR MOSKVA MUNOSABAT ORASIDA MUNOSABAT MOSKVA MUNOSABAT MOSKVA HUJUM MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT MOSKVA MUNOSABAT QIL MUNOSABAT BOSQICH QISQA RINBOSAR QISQA ORASIDA TAQDIM MUNOSABAT MOSKVA MUNOSABAT MOSKVA ORASIDA TAQDIM MUNOSABAT ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM ORASIDA TAQDIM']

# SHOULD BE
OPASINI ONASINI SOCH QISTIRGICHLARINI LABIGA QISTIRIB O YNAGANLAR XAYRLI TONG

**Comment by aslon1213** [comment]
Also there is another issue I noticed may be I am wrong. I tried to log the hyps in compute forward method in ASR class and strangely it is showing NONE

reference: 
```python
# Compute outputs
        hyps = None
        current_epoch = self.hparams.epoch_counter.current
        is_valid_search = (
            stage == sb.Stage.VALID
            and current_epoch % self.hparams.valid_search_interval == 0
        )
        is_test_search = stage == sb.Stage.TEST

        if is_valid_search:
            hyps, _, _, _ = self.hparams.valid_search(
                enc_out.detach(), wav_lens
            )

        elif is_test_search:
            hyps, _, _, _ = self.hparams.test_search(enc_out.detach(), wav_lens)
        logger.debug(f"HYPS: {hyps}")
```
[github_code_link ](https://github.com/speechbrain/speechbrain/blob/develop/recipes/CommonVoice/ASR/transformer/train.py 
)

**Comment by TParcollet** [comment]
Hi, you must show us the full yaml that you use with the inference class.

**Comment by aslon1213** [comment]
There was problem on my site sorry for this.... Thanks you for your help @TParcollet
